util/hash.h

```diff
@@ -49,4 +49,12 @@ struct SliceHasher {
   uint32_t operator()(const Slice& s) const { return GetSliceHash(s); }
 };
 
+// An alternative to % for mapping hashing to an arbitrary range. See
+// https://github.com/lemire/fastrange and
+// https://github.com/pdillinger/wormhashing/blob/2c4035a4462194bf15f3e9fc180c27c513335225/bloom_simulation_tests/foo.cc#L57
+inline uint32_t fastrange32(uint32_t a, uint32_t h) {
+  uint64_t product = (uint64_t)a * h;
+  return (uint32_t)(product >> 32);
```

Comment:
Nit: we follow Google C++ Style https://google.github.io/styleguide/cppguide.html#Casting, which requires us to use C++ style casting, like `static_cast<uint32_t>(product >> 32)`.